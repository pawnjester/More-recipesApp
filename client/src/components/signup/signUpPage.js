import React from 'react';
import '../../styles/signup.scss';
import Signupform from './signupform';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { userSignupRequest } from '../../actions/signupActions';



class signUpPage extends React.Component {
  render() {
    const {userSignupRequest} = this.props;
    
    return (
      <header id ="home-section">
      <div className= "dark-overlay">
      <div className= "home-inner">
        <div className ="container">
          <div className= "row justify-content-center">
            <div className='col-lg-6'>
              <div className= "card bg-white text-center card-form">
                <div className="card-body">
                  <h3>Sign Up</h3>
                  <p>Please fill this form to register</p>
                  <Signupform userSignupRequest = {userSignupRequest}/>
                  <div className="card-body ">                    
                    <p className="card-text text-right text-dark">Have an account? <a href="signin.html"  className="text-white">SIGN IN</a> </p>                    
                  </div>
                </div>
              </div>
            </div>
            
          </div>
          
        </div>
        
      </div>
      </div>
      
    </header>
    )
  }
}

signUpPage.propTypes = {
  userSignupRequest: PropTypes.func.isRequired,
}

export default connect(null, { userSignupRequest})(signUpPage)