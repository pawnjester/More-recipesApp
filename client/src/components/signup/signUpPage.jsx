import React from 'react';
import Signupform from './signupform';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { userSignupRequest } from '../../actions/signupActions';
import { addFlashMessage } from '../../actions/flashMessages';


class signUpPage extends React.Component {
  render() {
    const { userSignupRequest, addFlashMessage } = this.props;

    return (
      <header id="home-section">
        <div className="dark-overlay">
          <div className="home-inner">
          <div className="container">
          <div className="row justify-content-center">
            <div className="col-lg-6">
              <div className="card bg-white text-center card-form">
                <div className="card-body">
                  <h3>Sign Up</h3>
                  <p>Please fill this form to register</p>
                  <Signupform userSignupRequest={userSignupRequest} addFlashMessage={addFlashMessage} />
                  <div className="card-body ">
                    <p className="card-text text-right text-dark">Have an account? <a href="signin.html" className="text-danger">SIGN UP</a> </p>
                  </div>
                </div>
              </div>
            </div>

          </div>

        </div>

        </div>
        </div>

      </header>
    );
  }
}

signUpPage.propTypes = {
  userSignupRequest: PropTypes.func.isRequired,
  addFlashMessage: PropTypes.func.isRequired,

};

export default connect(null, { userSignupRequest, addFlashMessage })(signUpPage);

