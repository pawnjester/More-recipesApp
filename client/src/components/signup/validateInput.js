import isEmpty from 'lodash/isEmpty';
import Validator from 'validator';
/**
 * @description validation function
 *
 * @export
 *
 * @param {any} data
 *
 * @returns {void}
 */
export default function validateInput(data) {
  const errors = {};

  if (Validator.isEmpty(data.email)) {
    errors.email = 'This field is required';
  }
  if (!Validator.isEmail(data.email)) {
    errors.email = 'Email is Invalid';
  }

  if (Validator.isEmpty(data.username)) {
    errors.username = 'This field is required';
  } else if (data.username.length < 6) {
    errors.username = 'Username is too short, Must be min. of 6';
  }

  if (Validator.isEmpty(data.password)) {
    errors.password = 'This field is required';
  } else if (data.password.length < 6) {
    errors.password = 'Password is too short, Must be min. of 6 ';
  }

  if (Validator.isEmpty(data.passwordConfirmation)) {
    errors.passwordConfirmation = 'This field is required';
  }

  if (!Validator.equals(data.password, data.passwordConfirmation)) {
    errors.passwordConfirmation = 'Password must match';
  }

  return {
    errors,
    isValid: isEmpty(errors)
  };
}
