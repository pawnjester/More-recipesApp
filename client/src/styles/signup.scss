body{
        background:#fff;
        color: #fff;
        }      
        
      #home-section{
        min-height: 700px;
        background-size: cover;
        background-attachment: fixed;
      } 
      .dark-overlay{
        
        position: absolute;
        top: 0;
        left: 0;
        width: 100%;
        min-height: 700px; 
      } 
      .home-inner{
        padding-top: 100px; 
      }