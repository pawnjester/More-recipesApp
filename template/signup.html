<!DOCTYPE html>
<html lang = "en">
<head>
  <title>Sign in</title>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, shrink-to-fit=no">
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta/css/bootstrap.min.css" integrity="sha384-/Y6pD6FV/Vv2HJnA6t+vslU6fwYXjCFtcEpHbNJ0lyAFsXTsjBbfaDjzALeQsN6M" crossorigin="anonymous">
  <script src="https://use.fontawesome.com/a42b7208df.js"></script>

  <style>
    body{
        background:#fff;
        color: #fff;
        }
      
      .navbar{
        border-bottom: #FF4500 3px solid;
        opacity: 0.8;
        
      }    

      #home-section{
        min-height: 700px;
        background-size: cover;
        background-attachment: fixed;
      } 

      .dark-overlay{
        
        position: absolute;
        top: 0;
        left: 0;
        width: 100%;
        min-height: 700px; 
      } 

      .home-inner{
        padding-top: 100px; 
      }

      .card-form{
        opacity: 0.8;
      }  

  
  
  </style>
</head>
<body>
  <nav class ="navbar navbar-expand-sm bg-dark navbar-dark fixed-top">
      <div class= "container">
        <a href="index.html" class="navbar-brand">More-Recipes</a>            
                
      </div>     
      
  </nav>
   <header id ="home-section">
      <div class= "dark-overlay">
      <div class= "home-inner">
        <div class ="container">
          <div class= "row justify-content-center">
            <div class='col-lg-6'>
              <div class= "card bg-danger text-center card-form">
                <div class="card-body">
                  <h3>Sign In</h3>
                  <p>Please fill this form to register</p>
                  <form>
                    <div class= "form-group">
                      <input type="email" class="form-control form-control-lg" placeholder="Email"/>
                      
                    </div>
                    
                    <div class= "form-group">
                      <input type="text" class="form-control form-control-lg" placeholder="Username"/>
                      
                    </div>
                    <div class= "form-group">
                      <input type="password" class="form-control form-control-lg" placeholder="Password"/>
                      
                    </div>
                    <div class= "form-group">
                      <input type="password" class="form-control form-control-lg" placeholder="Confirm Password"/>
                      
                    </div>
                    
                    <input type="submit" value ="Submit" class="btn btn-outline-light btn-block">
                  </form>
                </div>
              </div>
            </div>
            
          </div>
          
        </div>
        
      </div>
      </div>
      
    </header>
</body>