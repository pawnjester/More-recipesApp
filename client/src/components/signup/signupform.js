import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import validateInput from './validateInput';
import TextFieldGroup from '../common/TextFieldGroup';
import {Redirect} from  'react-router-dom'

class signupform extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      email: '',
      password: '',
      passwordConfirmation: '',
      errors: {},
      isLoading: false,
      redirect: false
    }
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value})
  }

  isValid() {
    const { errors, isValid } = validateInput(this.state);

    if (!isValid) {
      this.setState({ errors });
    }

    return isValid;
  }

  onSubmit(e) {
    e.preventDefault();

    if (this.isValid()) {
    this.setState({errors: {}, isLoading: true});
    
    this.props.userSignupRequest(this.state).then(
      // () => {
      //   this.props.addFlashMessage({
      //     type: 'success',
      //     text: 'You have signed up successfully. Welcome!'
      //   })
        this.setState({redirect : true}),
              (err) => this.setState({ errors: err.response.data, isLoading: false })

    );
    }
  }
  render() {
    const {errors, redirect} = this.state;
    if (redirect) {
       return <Redirect to='/signin'/>;
     }
    return (
                <form onSubmit = {this.onSubmit}>

                  <TextFieldGroup
                    error = {errors.username || errors.errorname}
                    onChange = {this.onChange}
                    value = {this.state.username}
                    field = "username"
                    placeholder = "username" 
                    autofocus required
                  />

                  <TextFieldGroup
                    error = {errors.email || errors.errormail}
                    onChange = {this.onChange}
                    value = {this.state.email}
                    field = "email"
                    placeholder = "email"
                    
                  />

                  <TextFieldGroup
                    error = {errors.password}
                    onChange = {this.onChange}
                    value = {this.state.password}
                    field = "password"
                    placeholder = "password"
                    
                  />

                  <TextFieldGroup
                    error = {errors.passwordConfirmation}
                    onChange = {this.onChange}
                    value = {this.state.passwordConfirmation}
                    field = "passwordConfirmation"
                    placeholder = "Confirm Password"
                  />
                                      
                    
                    <input 
                    disabled = {this.state.isLoading}
                    type="submit"
                    value ="Submit"
                    className="btn btn-outline-light btn-block text-dark"
                    style ={{cursor: 'pointer'}} />
                  </form>
    )
  }
}

signupform.propTypes = {
  userSignupRequest: PropTypes.func.isRequired,
  // addFlashMessage: PropTypes.func.isRequired
}
export default signupform;

