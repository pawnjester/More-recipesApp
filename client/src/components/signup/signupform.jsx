import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import classnames from 'classnames';
import validateInput from './validateInput';
import TextFieldGroup from '../common/TextFieldGroup';

class signupform extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      email: '',
      password: '',
      passwordConfirmation: '',
      errors: {},
      isLoading: false,
      redirect: false,
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }


  onSubmit(e) {
    e.preventDefault();

    if (this.isValid()) {
      this.setState({ errors: {}, isLoading: true });

      this.props.userSignupRequest(this.state).then(
        this.setState({ redirect: true }),
        err => this.setState({ errors: err.response.data, isLoading: false }),

      );
    }
  }


  isValid() {
    const { errors, isValid } = validateInput(this.state);

    if (!isValid) {
      this.setState({ errors });
    }

    return isValid;
  }
  render() {
    const { errors, redirect } = this.state;
    if (redirect) {
      return <Redirect to="/signin" />;
    }
    return (
      <form onSubmit={this.onSubmit}>

        <TextFieldGroup
          error={errors.username || errors.errorname}
          onChange={this.onChange}
          value={this.state.username}
          field="username"
          placeholder="username"
          autofocus
          required
        />

        <TextFieldGroup
          error={errors.email || errors.errormail}
          onChange={this.onChange}
          value={this.state.email}
          field="email"
          placeholder="email"
        />

        <TextFieldGroup
          error={errors.password}
          onChange={this.onChange}
          value={this.state.password}
          field="password"
          placeholder="password"
          type="password"
        />

        <TextFieldGroup
          error={errors.passwordConfirmation}
          onChange={this.onChange}
          value={this.state.passwordConfirmation}
          field="passwordConfirmation"
          placeholder="Confirm Password"
          type="password"
        />


        <input
          disabled={this.state.isLoading}
          type="submit"
          value="Submit"
          className="btn btn-outline-light btn-block text-dark"
          style={{ cursor: 'pointer' }}
        />
      </form>
    );
  }
}

signupform.propTypes = {
  userSignupRequest: PropTypes.func.isRequired,
};
export default signupform;
