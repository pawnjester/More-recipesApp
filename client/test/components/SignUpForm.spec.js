import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';
import { SignUpForm } from '../../src/components/signup/SignUpForm';

configure({ adapter: new Adapter() });

const props = {
  userSignupRequest: jest.fn(() => Promise.resolve())
};
const setup = () => {
  const shallowWrapper = shallow(<SignUpForm {...props} />);
  return {
    shallowWrapper,
  };
};

describe('Sign Up Form', () => {
  it('should render correctly', () => {
    const { shallowWrapper } = setup();
    expect(shallowWrapper).toMatchSnapshot();
  });
});

describe('onChange() should', () => {
  it('be called when form input changes', () => {
    const { shallowWrapper } = setup();
    const event = {
      target: { name: 'username', value: 'randomxxcom' }
    };
    const inputForm = shallowWrapper.find('#username-field');
    inputForm.simulate('change', event);
    expect(shallowWrapper.instance().state.username).toBe('randomxxcom');
  });
});

describe('onSubmit() should', () => {
  it('be called when form is submitted', () => {
    const { shallowWrapper } = setup();
    const event = {
      preventDefault: jest.fn()
    };
    shallowWrapper.setState({
      username: 'kjdjkskjsd',
      email: 'kjfdkj@exa.com',
      password: 'jhdjsdkjds',
      passwordConfirmation: 'jhdjsdkjds',
    });
    shallowWrapper.instance().onSubmit(event);
    expect(props.userSignupRequest.mock.calls.length).toEqual(1);
  });
});
