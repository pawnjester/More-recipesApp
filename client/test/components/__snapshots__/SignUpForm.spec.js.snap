// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign Up Form should render correctly 1`] = `
<form
  id="Sign-Up-Form"
  onSubmit={[Function]}
>
  <TextFieldGroup
    autofocus={true}
    error=""
    field="username"
    id="username-field"
    onChange={[Function]}
    placeholder="username"
    required={true}
    type="text"
    value=""
  />
  <TextFieldGroup
    error=""
    field="email"
    onChange={[Function]}
    placeholder="email"
    type="text"
    value=""
  />
  <TextFieldGroup
    error=""
    field="password"
    onChange={[Function]}
    placeholder="password"
    type="password"
    value=""
  />
  <TextFieldGroup
    error=""
    field="passwordConfirmation"
    onChange={[Function]}
    placeholder="Confirm Password"
    type="password"
    value=""
  />
  <input
    className="btn btn-outline-light btn-block text-dark"
    disabled={false}
    id="submit-button"
    style={
      Object {
        "cursor": "pointer",
      }
    }
    type="submit"
    value="Submit"
  />
</form>
`;
